### ISSUE TITLE
AccountDataQuery return type consists of a vector of different types (Rewards, TXReceipts) and a scalar (AccountData)

### ISSUE THREAD

**Issue by sidsankhe** [spacemeshos/api#131]
Currently, the AccountDataQuery returns a vector which consists of a concatenation of the results of fetching Rewards, TransactionReceipts (which are to be implemented) and the AccountData. However, the AccountData is a scalar (always just one value). Furthermore, this makes offset handling within the database functions very tricky, since the offset now depends on how many results were returned by the other types. For example, if we are returning both TransactionReceipts and Rewards, then the offsets for the Rewards will depend not only on how many TransactionReceipts were returned, but could also depend on how many were present in the database as a whole (in case we need to skip over all of them)

**Comment by avive** [comment]
So what's the proposed change to the api design to address this issue?

**Comment by lrettig** [comment]
See: https://github.com/spacemeshos/api/blob/master/proto/spacemesh/v1/global_state_types.proto

I think we should split this endpoint into two or more endpoints. `max_results` is not really clearly defined when the response includes different data types, especially when those data types aren't intrinsically related (e.g., txs and rewards). This works better as a stream than as a query.

**Comment by avive** [comment]
I agree but is it for 0.2 or for 0.3? AccountDataQuery is the main way clients are going to use the api...

**Comment by lrettig** [comment]
The endpoint works, it could just be better designed. This is a breaking change so it should go into API 2.0 which I guess means milestone 0.3.
